pyinteraph/path_analysis.py

```diff
@@ -0,0 +1,577 @@
+                # Add edge
+                graph.add_edge(node1, node2, e_weight = inc)
+    return graph
+
+def filter_graph(graph, node_threshold, edge_threshold):
+    """Takes in a graph, and returns a graph which only contains nodes
+    and edges from the provided graph that have a weight higher than
+    the provided threshold.
+    """
+
+    filterd_graph = nx.Graph()
+    for u, v, d in graph.edges(data = True):
+        # Check that both thresholds are met
+        if graph.nodes()[u]["n_weight"] > node_threshold and \
+           graph.nodes()[v]["n_weight"] > node_threshold and \
+           d["e_weight"] > edge_threshold:
+            # Add node weights first
+            filterd_graph.add_node(u, n_weight = graph.nodes()[u]["n_weight"])
+            filterd_graph.add_node(v, n_weight = graph.nodes()[v]["n_weight"])
+            # Add edge weight
+            filterd_graph.add_edge(u, v, e_weight = d["e_weight"])
+    # Warn if no edges in metapath graph
+    if nx.is_empty(filterd_graph):
+        log.warning("No metapaths found.")
+    return filterd_graph
+
+def get_metapath(graph, res_id, res_space, node_threshold, edge_threshold):
+    """Takes in a PSN graph where weights are persistence values. Returns
+    a graph of the metapath
+    """
+    
+    # Calculate all shortest paths
+    paths = get_all_shortest_paths(graph, res_id, res_space)
+    # Create graph from path list
+    paths_graph = get_graph_from_paths(paths)
+    # Filter graph
+    metapath_graph = filter_graph(paths_graph, node_threshold, edge_threshold)
+    return metapath_graph
+
+def plot_graph(fname, graph, hub_num, col_map_e, col_map_n, dpi):
+    """Takes in a graph and saves a png of the plot. Also takes in a hub
+    cutoff value. Nodes with a larger number of edges than the cutoff
+    are highlighted.
+    """
+    # Get attributes
+    weights = [d["e_weight"] for u, v, d in graph.edges(data=True)]
+    nodes = np.array([n for n, d in graph.degree(graph.nodes())])
+    degrees = np.array([d for n, d in graph.degree(graph.nodes())])
+    selection = degrees >= hub_num
+    hubs = nodes[selection]
+    hubs_deg = degrees[selection]
+    non_hubs = nodes[np.logical_not(selection)]
+    non_hubs_deg = degrees[np.logical_not(selection)]
+    unique_hubs = len(np.unique(hubs_deg))
+    # Get positions. Larger k values make the nodes spread out more
+    pos = nx.spring_layout(graph, k=0.2, iterations=30)
+    # Get cmaps
+    # Gray scale for nodes (select how many greys to pick)
+    node_colors = sns.color_palette(col_map_n, max(degrees) - hub_num + 1)
+    cmap_n = mplcolors.LinearSegmentedColormap.from_list('node_colors', 
+                                                   node_colors, 
+                                                   N = len(node_colors))
+    # Color palette for edges
+    edge_colors = sns.color_palette(col_map_e)
+    cmap_e = mplcolors.LinearSegmentedColormap.from_list('edge_colors', 
```

Comment:
does this need to be segmented? I think it can just be continuous

Reply:
I originally tried to use `sns.color_palette("rocket", as_cmap=True)`  (I got this from the seaborn website) but this resulted in an error saying `as_cmap` is an unexpected input which doesn't make sense since clearly it's on their website and I couldn't find a fix by googling the error. That's why I used the discrete palette. An alternative that works is `mplcolors.ListedColormap(sns.color_palette("rocket_r", 256), name = 'edge_colors'))` but if I understand correctly, this is also discrete, so this isn't what you want? I'm not sure how else to use a continuous map.

Reply:
Sorry - by continuous I actually meant a `ListedColormap`, which as you mention isn't really continuous. Just it seemed to be suitable for cases in which you don't want to explicitly segment the color map.

I'm not sure why the first line you quoted wouldn't work - works on my virtualenv (seaborn 0.11.0).

About the second alternative, why `rocket_r`? Isn't it the opposite respect to `rocket`?

Reply:
I have seaborn 0.9.1 so I guess that could've been a problem. Sorry, the original should've also been `rocket_r`, I just forgot to add it while testing.
Is the current `ListedColormap` okay or should I update seaborn and try to change it?